File: 198. House Robber.cpp
// 198. House Robber

class Solution {
public:
    int rob(vector<int>& nums) {
        int rob1 = 0, rob2 = 0;
        for (int n : nums) {
            int temp = max(rob1 + n, rob2);
            rob1 = rob2;
            rob2 = temp;
        }
        return rob2;
    }
};

File: 198. House Robber.py
# 198. House Robber


class Solution:
    def rob(self, nums: List[int]) -> int:
        rob1, rob2 = 0, 0

        for n in nums:
            temp = rob1 + n
            rob1 = rob2
            rob2 = max(temp, rob2)
        return rob2